transport/src/test/java/io/netty/channel/CoalescingBufferQueueTest.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.channel;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.concurrent.ImmediateEventExecutor;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.MockitoAnnotations;
+
+import java.nio.charset.Charset;
+
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link CoalescingBufferQueue}.
+ */
+public class CoalescingBufferQueueTest {
+
+    private ByteBuf cat = Unpooled.wrappedBuffer("cat".getBytes(Charset.defaultCharset()));
+    private ByteBuf mouse = Unpooled.wrappedBuffer("mouse".getBytes(Charset.defaultCharset()));
```

Comment:
@louiscryan - Should we initialize these in `@Before` so the lifetime is more clear?  They are setup before a test, and released after a test.